src/core/graph/graph.ts

```
@@ -0,0 +1,290 @@
+
+  /**
+   * Add a station to the spatial index
+   */
+  private addToSpatialIndex(stationId: string, coordinates: Coordinates): void {
+    const cellKey = getCellKey(coordinates);
+
+    if (!this.spatialIndex.has(cellKey)) {
+      this.spatialIndex.set(cellKey, []);
+    }
+
+    const stationIds = this.spatialIndex.get(cellKey)!;
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)